docs/reference/mpremote.rst

```diff
@@ -59,84 +95,163 @@ The full list of supported commands are:
+  want to run a subsequent command on a board without first soft-resetting it.
+
+.. _mpremote_command_soft_reset:
 
-- perform a soft-reset of the device:
+- **soft-reset** -- perform a soft-reset of the device:
 
   .. code-block:: bash
 
       $ mpremote soft-reset
 
   This will clear out the Python heap and restart the interpreter.  It also
-  disables auto soft-reset.
+  prevents the subsequent command from triggering :ref:`auto-soft-reset <mpremote_reset>`.
 
-- enter the REPL on the connected device:
+.. _mpremote_command_repl:
+
+- **repl** -- enter the REPL on the connected device:
 
    .. code-block:: bash
 
-      $ mpremote repl [options]
+      $ mpremote repl [--options]
 
   Options are:
 
   - ``--escape-non-printable``, to print non-printable bytes/characters as their hex code
   - ``--capture <file>``, to capture output of the REPL session to the given
     file
   - ``--inject-code <string>``, to specify characters to inject at the REPL when
-    Ctrl-J is pressed
+    ``Ctrl-J`` is pressed. This allows you to automate a common command.
   - ``--inject-file <file>``, to specify a file to inject at the REPL when
-    Ctrl-K is pressed
+    ``Ctrl-K`` is pressed. This allows you to run a file (e.g. containing some
+    useful setup code, or even the program you are currently working on).
+
+  While the ``repl`` command running, you can use ``Ctrl-]`` or ``Ctrl-x`` to
+  exit.
+
+  **Note:** The name "REPL" here reflects that the common usage of this command
+  to access the Read Eval Print Loop that is running on the MicroPython
+  device. Strictly, the ``repl`` command is just functioning as a terminal
+  (or "serial monitor") to access the device. Because this command does not
+  trigger the :ref:`auto-reset behavior <mpremote_reset>`, this means that if
+  a program is currently running, you will first need to interrupt it with
+  ``Ctrl-C`` to get to the REPL, which will then allow you to access program
+  state. You can also use ``mpremote soft-reset repl`` to get a "clean" REPL
+  with all program state cleared.
 
-- evaluate and print the result of a Python expression:
+.. _mpremote_command_eval:
+
+- **eval** -- evaluate and print the result of a Python expression:
 
   .. code-block:: bash
 
       $ mpremote eval <string>
 
-- execute the given Python code:
+.. _mpremote_command_exec:
+
+- **exec** -- execute the given Python code:
 
   .. code-block:: bash
 
       $ mpremote exec <string>
 
-- run a script from the local filesystem:
+  By default, ``mpremote exec`` will display any output from the expression until it
+  terminates. The ``--no-follow`` flag can be specified to return immediately and leave
+  the device running the expression in the background.
+
+.. _mpremote_command_run:
+
+- **run** -- run a script from the local filesystem:
 
   .. code-block:: bash
 
-      $ mpremote run <file>
+      $ mpremote run <file.py>
+
+  This will copy the file to RAM on the device and execute it directly.
```

Comment:
I'm not sure about the suggested wording because to me "transfer to the device" implies copying it to the filesystem. I've updated the wording to not say "copy". (And sentence about what this command is useful for).